crates/provider/src/provider/trait.rs

```diff
@@ -534,6 +534,50 @@ pub trait Provider<T: Transport + Clone = BoxTransport, N: Network = Ethereum>:
         self.client().request("eth_getTransactionByHash", (hash,)).into()
     }
 
+    /// Returns the transaction by block hash and transaction index position.
```

Comment:
nit: change the docs to be in line with the other ones, e.g. "Get a transaction by a block number and its position in the block"